cpp/ql/lib/semmle/code/cpp/models/interfaces/PartialFlow.qll

```diff
@@ -0,0 +1,28 @@
+/**
+ * Provides an abstract class to override the implicit assumption that a
+ * dataflow/taint-tracking model always fully override the parameters they are
+ * are modeled as writing to. To use this QL library, create a QL class
+ * extending `PartialFlowFunction` with a characteristic predicate that selects
+ * the function or set of functions you are modeling and override the
+ * `isPartialWrite` predicate.
+ *
+ * Note: Since both `DataFlowFunction` and `TaintFunction` extend this class
+ * you don't need to explicitly add this as a base class if your QL class
+ * already extends either `DataFlowFunction` or `TaintFunction`.
+ */
+
+import semmle.code.cpp.Function
+import FunctionInputsAndOutputs
+import semmle.code.cpp.models.Models
+
+/**
+ * A function that may (but not always) updates (part of) a `FunctionOutput`.
```

Comment:
Good idea. I've done both of those suggestions in 671904d58c5c9578497ab2bbd235173484543c7e (I couldn't accept the suggestion because "The diff is outdated" according to GitHub)

Reply:
Excellent, don't worry about the suggestion.